stream_alert/rule_processor/threat_intel.py

```diff
@@ -13,90 +13,389 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 """
-import csv
-import gzip
-import os
+import backoff
+import boto3
+from boto3.dynamodb.types import TypeDeserializer
+from botocore.exceptions import ClientError
+
+from stream_alert.shared import NORMALIZATION_KEY
+from stream_alert.shared.backoff_handlers import (
+    backoff_handler,
+    success_handler,
+    giveup_handler
+)
 
 from stream_alert.rule_processor import LOGGER
 
+
+# DynamoDB Table settings
+MAX_QUERY_CNT = 100
+PRIMARY_KEY = 'ioc_value'
+SUB_TYPE_KEY = 'sub_type'
+PROJECTION_EXPRESSION = '{},{}'.format(PRIMARY_KEY, SUB_TYPE_KEY)
+
+class StreamIoc(object):
+    """Class to store IOC info"""
+    def __init__(self, **kwargs):
+        """Initialize StreamIoc instance and store useful information
+
+        Keyword arguments:
+            - value (str): IOC value
+            - ioc_type (str): Type of IOC, 'domain', 'ip' or 'md5'
+            - associated_record (dict): Original record carry IOC value.
+            - is_ioc (bool): Indicate is the value in DynamoDB IOC table or not. If
+                True, it means it is malicious IOC, otherwise it is False (default)
+        """
+        self._value = kwargs.get('value', None)
+        self._ioc_type = kwargs.get('ioc_type', None)
+        self._associated_record = kwargs.get('rec', None)
+        self._is_ioc = kwargs.get('is_ioc', False)
+
+    @property
+    def value(self):
+        """Get the value of an IOC instance"""
+        return self._value
+
+    @value.setter
+    def value(self, value):
+        """Set the value of an IOC instance"""
+        self._value = value
+
+    @property
+    def ioc_type(self):
```

Comment:
Not that I'm not proud of you for the pythonic use of `@property` 😄 .. the getter/setter approach is most useful if you must transform the data being set/got before setting/returning it (ie: transforming a string, adding 100 to a base integer, etc). In this case, you're actually just adding unnecessary complexity. You could just have the instance properties declared as public, avoid all of the extra code, and have the same affect:

```
def __init__(self, **kwargs):
    self.value = kwargs.get('value', None)
    self.ioc_type = kwargs.get('ioc_type', None)
    self.associated_record = kwargs.get('rec', None)
    self.is_ioc = kwargs.get('is_ioc', False)
```